program/FORCE-RISCV ISG preliminary project proposal.md

```diff
@@ -0,0 +1,145 @@
+The present document gives a present snapshot of requirements and Preliminary Project Proposal.  More detailed requirements will be available with Project Proposal.
+
+### Introduction
+32-bit RISC-V support and additional new features for both 64-bit and 32-bit RISC-V.
+
+### Initial project requirements 
+1. Support for 32-bit RISC-V instructions.
+2. Support for 32-bit RISC-V registers.
+3. Support for 32-bit RISC-V virtual memory system.
+4. Support for 32-bit RISC-V exception handling.
+5. Support for 32-bit RISC-V front-end framework.
+6. Support 32-bit RISC-V ISS standalone and shared library.
+7. MP/MT support for both 32-bit and 64-bit RISC-V.
+8. Advanced resource dependency for both 32-bit and 64-bit RISC-V.
+
+### Future enhancements:
+9. Support for new RISC-V extensions and updates.
+10. New features requested by OpenHW Group memebers and broader open source community.
+
+## Explanation of why OpenHW should do this project
+
+The following strengths of FORCE-RISCV are in comparison to other open source ISGs where FORCE-RISCV does it better, if we implement support for 32-bit RISC-V, your 32-bit RISC-V design will also benefit from the high quality test generation.
+      �    High degree of test configurability via extensive Python front-end APIs.
+      �    High quality of randomness with generate-and-step using the linked ISS library, able to achieve high density of interesting test features and reach high micro-architecture coverage requirements.
+      �    Generates ELF files directly, no dependency on external toolchain compilation, simplifies the simulation work-flow.
+      �    Great test generation speed, able to scale well for heavy MP/MT processor verification.  Due to the origin of FORCE-RISCV from server-class CPU projects.
+
+Additional advanced features like MP/MT and resource dependency will meet the need of more complicated processor design that OpenHW Group might tackle in the future.
+
+## Industry landscape: description of competing, alternative, or related efforts in the industry
+
+RISCV-DV from Google is currently a very popular test generator for RISC-V designs.
+
+### Related efforts to be described
+- Google's RISCV-DV released in Jan 23, 2019, 1 year and half before Futurewei Technologies' FORCE-RISCV open source and there is no other open source choices availabe.
+- RISCV-DV has all/most basic functionalities that a open source RISC-V 64-bit/32-bit processor project would need.
+- RISCV-DV lacks highly fine-grained control that FORCE-RISCV provide and due to not emulating generated instructions test generated are much too coarse to reach high coverage as the processor design get more complicated.
+- RISCV-DV would be not possible to be extended further in features beyond certain point.
+- Dependency on (customized) GCC toolchain.
+- Needs commercial license for System Verilog compile/simulating tools to contribute.
+- OpenHW Group don't have much influence on the project direction.

```

Comment:
You can say "- OpenHW Group doesn't currently have involvement in the RISC-DV project direction".

Reply:
Well put.